src/sharp_plugin.c

```diff
@@ -122,8 +123,8 @@ int ncclSharpAllGather(void *context, void *buf, int len) {
 
   p2p_plugin = nccl_p2p_get_plugin_type();
   if (p2p_plugin != NCCL_P2P_UCX) {
-    NCCLCHECK(NCCL_PLUGIN_SYMBOL.regMr(cComm->recvComm, buf, cComm->nranks*len, NCCL_PTR_HOST, &rMhandle));
-    NCCLCHECK(NCCL_PLUGIN_SYMBOL.regMr(cComm->sendComm, buf, cComm->nranks*len, NCCL_PTR_HOST, &sMhandle));
+    NCCLCHECK(ncclNetPlugin_v6.regMr(cComm->recvComm, buf, cComm->nranks*len, NCCL_PTR_HOST, &rMhandle));
```

Comment:
I'm not sure this change is needed. It is fine to keep NCCL_PLUGIN_SYMBOL to ease maintenance over time. You'll need to change this line only if regMr changes.